File: backend/features/product/product.model.js
const mongoose = require('mongoose')

class Product {
  constructor() {
    this._name = 'product'
  }

  _init() {
    const schema = new mongoose.Schema(
      {
        name: {
          type: String,
          required: true,
        },
        image: {
          type: String,
          required: true,
        },
      },
      { timestamps: true }
    )

    mongoose.model(this._name, schema)
  }

  getInstance() {
    try {
      return mongoose.model(this._name)
    } catch (err) {
      this._init()
    }

    return mongoose.model(this._name)
  }

  get name() {
    return this._name
  }
}

module.exports = Product

File: kalam_automation/lib/models/product.model.dart
import 'package:flutter/foundation.dart';

class Product {
  final String id;
  final String name;
  final String image;

  const Product({
    @required this.id,
    @required this.name,
    @required this.image,
  });

  factory Product.fromJson(dynamic product) {
    return Product(
      id: product['_id'],
      name: product['name'],
      image: product['image'],
    );
  }
}
